### PR TITLE
KOGITO-5676: BPMN Editor - Containment not working when Node overlaps the Connector while splicing

### PR DESCRIPTION
**Pull request by handreyrc**
Hey @romartin, please could you review this PR?

**JIRA**: [KOGITO-5676](https://issues.redhat.com/browse/KOGITO-5676)

**VS Code**: [plugin](https://drive.google.com/file/d/1yGvvu7NSdkIwhpgRHqrcYDOdXO4VF7uV/view?usp=sharing)

Thanks

### CODE REVIEW

**Review by romartin — APPROVED** [review]
Looks good to me, thanks @handreyrc !

**Comment by handreyrc** [comment]
Hey @romartin @LuboTerifaj,
I fixed a small side-effect, added more unit tests, rebased and updated the plugin.  This PR is ready for review and testing again.

Thanks!

**Comment by LuboTerifaj** [comment]
Hello @handreyrc , VSCode extension is missing in both google drive you linked here and the build in PR's checks. Can you please have a look at it? Thanks!

**Review by LuboTerifaj — COMMENTED** [review]
Hello @handreyrc , 

I couldn't have reproduce the issue on main branch. Could you please add the steps to reproduce to the jira? See also my [comment](https://issues.redhat.com/browse/KOGITO-5676?focusedCommentId=18854146&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-18854146) there.

Thank you!

**Comment by handreyrc** [comment]
Hey @LuboTerifaj,
It's strange you could,t get the vscode plugin from my link, maybe some browser cache issue. 
Anyway here goes a new link for the plugin:

**VS Code**: [plugin](https://drive.google.com/file/d/1yGvvu7NSdkIwhpgRHqrcYDOdXO4VF7uV/view?usp=sharing) 

I already updated the issue with the steps to reproduce it as well, please ping me if you can't reproduce it yet.

The FDB is failing due to some unrelated issue, I'll keep an eye to get it green.

Thanks for checking it!

**Review by karreiro — APPROVED** [review]
Thank you, @handreyrc! Great stuff :-)

**Comment by handreyrc** [comment]
Hey @LuboTerifaj we have all green here.
I tried to answer your questions in the JIRA the best I could I hope it helps.
I tested the lastest main and I could reproduce the issue. 
Please ping me if you can't reproduce it yet and we can do a quick call.

Thanks!

File: lienzo-tests/src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresLineSpliceControlImplTest.java [hunk 1]
```diff
@@ -645,6 +647,66 @@ public void testExecuteSpliceWithContainment() {
 
         // Check containment after
         assertEquals(1, parent.getChildShapes().size(), 0);
+        verify(index, never()).findShapeAt(WiresLineSpliceControlImpl.XY_OFFSET,
+                                           -WiresLineSpliceControlImpl.XY_OFFSET);
+    }
+
+    @Test
+    public void testExecuteSpliceWithContainmentWithDeviation() {
```
Comment (LuboTerifaj) [inline lienzo-tests/src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresLineSpliceControlImplTest.java:655]: There are many common lines of code in `testExecuteSpliceWithContainmentWithoutDeviation` and `testExecuteSpliceWithContainmentWithDeviation` test methods. Would it make sense to you to refactor it a little bit? You could create a method for common code for example.

**Review by LuboTerifaj — CHANGES_REQUESTED** [review]
Hello @handreyrc , I have updated the steps to reproduce in jira.

I didn't find any issues. The fix works as expected.

However, I have one comment regarding code.

I have found related issue regarding the sequence flow colour inside containers (present also on main branch), but I have reported it in separate jira:

- [KOGITO-5778](https://issues.redhat.com/browse/KOGITO-5778) Sequence flow is red during Line splicing inside containers

Thank you!

**Review by LuboTerifaj — APPROVED** [review]
Hello @handreyrc , The extension you have linked in the comment above works as expected. Thanks!

**Comment by romartin** [comment]
Hey @handreyrc , Can I merge this? I see the FDB failing, is it something related to this PR or we can go ahead? Thx

**Comment by handreyrc** [comment]
Hey @romartin,

This PR is rebased and finally has a green FDB. It is ready to merge!

Thanks!

**Pull request merged**